x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/grok.test.ts

```diff
@@ -126,4 +126,40 @@ describe('Processor: Grok', () => {
 
     expect(processors[0][GROK_TYPE].patterns).toEqual([escapedValue]);
   });
+
+  test('accepts pattern definitions that contains escaped characters', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+      find,
+      component,
+    } = testBed;
+
+    // Add "field" value
+    form.setInputValue('fieldNameField.input', 'test_grok_processor');
+
+    // Add pattern 1
+    form.setInputValue('droppableList.input-0', 'pattern1');
+
+    await act(async () => {
+      find('patternDefinitionsField').simulate('change', {
+        jsonContent: '{"pattern_1":"""aaa(bbb""", "pattern_2":"aaa(bbb"}',
```

Comment:
nit (applies to all test files): can we try using triple-quote strings that contain illegal characters (e.g. `"` or `\`) and check that they are correctly escaped when converted to single-quote chars? We can also add the equivalent single-quote chars and make sure they are not changed when being serialized.